### ISSUE TITLE
[Bounty] Port for Mac OS version of the game

### ISSUE THREAD

**Issue by Foul-Tarnished** [ersh1/BG3_NativeCameraTweaks#17]
Hi

Would you be interesting in porting this mod to the Mac OS version of BG3 ?
I imagine NativeModLoader would need to be ported too.

I'm willing to pay 100$ for NativeModLoader, and 100$ for NativeCameraTweaks.
I also proposed this to NativeModLoader author on Discord.

Thanks for your amazing mods ❤️

**Comment by Foul-Tarnished** [comment]
https://github.com/LionAG/BaldurGateIII-ImprovedCamera/issues/2

**Comment by Foul-Tarnished** [comment]
Maybe this can help inject code on MacOS
https://github.com/LIJI32/MIP

**Comment by milljm** [comment]
Reading more about this, I imagine SIP will prevent any sort of injection technique. Even if this mod supported Mac, but only if users disable [SIP](https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection)... I don't think I myself would use it.

Sadly, I think we must wait until Larian allows such "plugin" loading support.

I am heart warmed that you offered a bounty, and I was about to second it, as well as look into forking this project. But I don't it will be feasible.

**Comment by ersh1** [comment]
Yeah. It'd be nice to see a MacOS version, but I personally have no experience with MacOS and can't even judge whether that's possible. You can imagine there's no realistic bounty that could incentivize me to learn a whole different world that I have no personal interest in :) I'll keep this open, of course, in hopes that someone specialized comes by.

As for Larian's official mod support changing things in this case, I wouldn't hold my breath. The way native code plugins work makes any sort of official mod support irrelevant. They're still relying on hacking the executable, are very fragile when the game updates (without any special measures the mod would have to be updated every single time the executable is recompiled) and are inherently a security risk (a **big** no-no for something that would essentially be officially endorsed).

The only real way to officially support such a mod is to integrate it into the game's source code itself. The only examples of something like that happening that I know of, are games that weren't actively developed anymore, and had a willing and passionate community dev team that was given NDA'd access to the source code. This doesn't mean it couldn't happen, of course. Just unlikely.

**Comment by Foul-Tarnished** [comment]
Yeah it would require to disable SIP, which is a bummer

I doubt Larian modding tools will let us edit camera variables and all, or even inject code seamlessly and officially

**Comment by ersh1** [comment]
> NDA access to source code would be great, but not sure how that would even help for that. Recompile the game with these camera tweaks integrated directly ? That would be a mess for mod users (as you could only give them the .exe/.app ?), also in terms of security it wouldn't be great (malware in executable etc)

Like you mentioned, redistributing the whole exe wouldn't solve anything. What I meant was literally integrating the mod into the game. As in, it's no longer a mod, it's an optional setting included in every copy of the game. Anything else just isn't possible to support officially if the mod in question modifies the executable's code - as opposed to modifying some game files (meshes, textures, UI) or scripts.

**Comment by Foul-Tarnished** [comment]
I imagine the patch 7 toolkit doesn't help ? :/ @ersh1 
[Moonglasses](https://www.nexusmods.com/baldursgate3/mods/12308) let us unlock "hidden" features of the toolkit, so we can edit the world map, edit osiris scripts, etc.

Maybe this could be used to make a light version of this mod, by only editing osiris scripts ?
Though I'm not sure the camera is an osiris script...

**Comment by Foul-Tarnished** [comment]
So Patch 8 will have an official photo mode

Maybe it could help make a mod like this for Mac ?
Make the photo mode worl durinf exploration and combat 🙏
